pycrostates/preprocessing/spatial_filter.py

```diff
@@ -0,0 +1,190 @@
+      weights based on the reduced adjacency matrix.
+    * Reduced interpolation matrices are normalized.
+    * Channel's timepoints are interpolated
+      using their reduced interpolation matrix.
+
+    Parameters
+    ----------
+    inst : Raw | Epochs | ChData
+        Instance to spatial filter.
+    ch_type : str
+        The channel type on which to apply spatial filter.
+        Currently only supports ``'eeg'``.
+        Default to 'eeg'.
+    exclude_bads : bool
+        If set to ``True`` bad channels will be removed
+        from the adjacency matrix and therefore not be used
+        to interpolate neighbors. In addition, bad channels
+        will not be filtered.
+        If set to ``False``, proceed as if all channels were good.
+        Default to ``True``.
+    origin : array-like, shape (3,) | str
+        Origin of the sphere in the head coordinate frame and in meters.
+        Can be ``'auto'`` (default), which means a head-digitization-based
+        origin fit.
+    %(n_jobs)s
+    %(verbose)s
+
+    Returns
+    -------
+    inst : Raw | Epochs| ChData
+        The modified instance.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    # Checks
+    from ..io import ChData
+
+    _check_type(inst, (BaseRaw, BaseEpochs, ChData), item_name="inst")
+    _check_value(ch_type, ("eeg",), item_name="ch_type")
+    n_jobs = _check_n_jobs(n_jobs)
+    # check preload for Raw
+    _check_preload(inst, "Apply spatial filter")
+    # check montage
+    if inst.get_montage() is None:
+        raise ValueError(
+            "No montage was set on your data, but spatial filter"
+            "can only work if digitization points for the EEG "
+            "channels are available. Consider calling inst.set_montage() "
+            "to apply a montage."
+        )
+    # remove bad channels
+    # Extract adjacency matrix
+    adjacency_matrix, ch_names = mne.channels.find_ch_adjacency(
+        inst.info, ch_type
+    )
+    adjacency_matrix = adjacency_matrix.todense()
+    adjacency_matrix = np.array(adjacency_matrix)
+    if exclude_bads:
+        for c, chan in enumerate(ch_names):
+            if chan in inst.info["bads"]:
+                adjacency_matrix[c, :] = 0  # do not change bads
+                adjacency_matrix[:, c] = 0  # don't use bads to interpolate
+    # retrieve picks based on adjacency matrix
+    picks = dict(_picks_by_type(inst.info, exclude=[]))[ch_type]
+    info = mne.pick_info(inst.info, picks)
+    assert ch_names == info.ch_names
+    # retrieve channel positions
+    pos = inst._get_channel_positions(picks)
+    # test spherical fit
+    origin = _check_origin(origin, info)
+    distance = np.linalg.norm(pos - origin, axis=-1)
+    distance = np.mean(distance / np.mean(distance))
+    if np.abs(1.0 - distance) > 0.1:
+        logger.warn(
+            "Your spherical fit is poor, interpolation results are "
+            "likely to be inaccurate."
+        )
```

Comment:
I'm not sure we need the 'origin' argument. It's only use is to do this check, and thus for a user, the only reason to change this argument would be to get rid of this warning. 
I'm not entirely sure about what this check does, but I fee like there must be a better way to handle this.